### PR TITLE
Clean up real device setup docs

### PR DESCRIPTION
**Pull request by imurchie**
## Proposed changes

The documentation for real device setup is old and not particularly useful anymore. Better documentation exists in `appium-xcuitest-driver` for that particular system. Move that here, and integrate it.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Fulfills #7577

### CODE REVIEW

File: docs/en/appium-setup/real-devices-ios.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+
+
+## Configuring iOS Real Device Tests: Xcode 8 and iOS 9.3 and above
+
+In addition to the dependency on `libimobiledevice`, Appium support for real
+devices running iOS 9.3 and above using Xcode 8+ also depends on `ios-deploy`,
+which is easily available through [npm](https://www.npmjs.com/package/ios-deploy)
+```
+npm install -g ios-deploy
```
Comment (jlipps) [inline docs/en/appium-setup/real-devices-ios.md]: unrelated to this PR, but if `ios-deploy` is on NPM, why don't we just include it in xcuitest-driver's deps and call out to it from there?

**Review by jlipps — APPROVED** [review]
this will make things much more transparent for users!

File: docs/en/appium-setup/real-devices-ios.md (hunk 1 above)
Comment (imurchie) [inline docs/en/appium-setup/real-devices-ios.md]: I feel like there was a reason. A really great reason. The best.

But I can't remember what it was at the moment.

Comment (jlipps) [inline docs/en/appium-setup/real-devices-ios.md]: Maybe we didn't want it to rebuild native bindings? dunno.

Comment (imurchie) [inline docs/en/appium-setup/real-devices-ios.md]: I just tried. I think that the issue was that it is a package that compiles on install. We could put it in optional dependencies, and change the wording here.

**Pull request merged**